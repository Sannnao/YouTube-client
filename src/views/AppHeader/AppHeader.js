export default class AppHeader {
  static render() {
    const header = document.createElement('header');
    const inputField = document.createElement('input');

    header.setAttribute('class', 'header');
    header.appendChild(inputField);

    inputField.setAttribute('class', 'input-field');

    document.body.appendChild(header);
  }
}
