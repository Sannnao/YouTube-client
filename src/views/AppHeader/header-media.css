@media (max-width: 1100px) {
  .input-field {
    width: 60%;
  }
}

@media (max-width: 700px) {
  .input-field {
    width: 70%;

  }
}

@media (max-width: 600px) {
  .input-field {
    width: 80%;

  }
}

@media (max-width: 500px) {
  .input-field {
    width: 90%;
  }
}
