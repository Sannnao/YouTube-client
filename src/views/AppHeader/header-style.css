.header {
  display: flex;
  justify-content: center;
  align-items: center;
}

.input-field {
  width: 50%;
  height: 30px;
  padding: 0 40px;
  font-family: inherit;
  font-size: 18px;
  background: #fff url('../../assets/icons/search.png') 10px 50%/20px no-repeat;
  border: none;
  border-radius: 20px;
  box-shadow: 0 1px 5px 0 rgb(190, 190, 190);
  outline: none;
}

.input-field:hover {
  box-shadow: 0 1px 5px 0 rgb(165, 165, 165);
}

.input-field:focus {
  box-shadow: 0 1px 5px 0 rgb(165, 165, 165), inset 0 0 1px 1px rgb(165, 165, 165);
}
