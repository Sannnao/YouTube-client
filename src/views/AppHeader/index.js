import AppHeader from './AppHeader';
import './header-style.css';
import './header-media.css';

export default AppHeader;
