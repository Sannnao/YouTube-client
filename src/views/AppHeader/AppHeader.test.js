import AppHeader from './AppHeader';

describe('AppHeader.render', () => {
  it('Should be an instance of Function', () => {
    expect(AppHeader.render).toBeInstanceOf(Function);
  });

  it('Should be render correctly', () => {
    AppHeader.render();
    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
