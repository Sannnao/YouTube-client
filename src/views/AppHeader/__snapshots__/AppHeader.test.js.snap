// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader.render Should be render correctly 1`] = `"<header class=\\"header\\"><input class=\\"input-field\\"></header>"`;
